omero/annotation_scripts/Export_to_csv.py

```diff
@@ -0,0 +1,582 @@
+                    rstring("- Dataset"), rstring("-- Image"),
+                    rstring("Screen"), rstring("- Plate"),
+                    rstring("-- Well"), rstring("-- Acquisition"),
+                    rstring("--- Image")
+    ]
+
+    separators = [";", ",", "TAB"]
+    # Here we define the script name and description.
+    # Good practice to put url here to give users more guidance on how to run
+    # your script.
+    client = scripts.client(
+        'Export to CSV',
+        """
+    This script exports for the selected objects their name, IDs and associated
+    key-value pairs.
+    \t
+    Check the guide for more information on parameters and errors:
+    https://guide-kvpairs-scripts.readthedocs.io/en/latest/index.html
+    \t
+    Default namespace: openmicroscopy.org/omero/client/mapAnnotation
+        """,  # Tabs are needed to add line breaks in the HTML
+
+        scripts.String(
+            P_DTYPE, optional=False, grouping="1",
+            description="Parent data type of the objects to annotate.",
+            values=source_types, default="Dataset"),
+
+        scripts.List(
+            P_IDS, optional=False, grouping="1.1",
+            description="List of parent data IDs containing the objects " +
+                        "to delete annotation from.").ofType(rlong(0)),
+
+        scripts.String(
+            P_TARG_DTYPE, optional=True, grouping="1.2",
+            description="Choose the object type to delete annotation from.",
+            values=target_types, default="<on current>"),
+
+        scripts.List(
+            P_NAMESPACE, optional=True,
+            grouping="1.3",
+            description="Namespace(s) to include for the export of key-" +
+                        "value pairs annotations. Default is the client" +
+                        "namespace, meaning editable in " +
+                        "OMERO.web").ofType(rstring("")),
+
+        scripts.Bool(
+            "Other parameters", optional=True, grouping="2", default=True,
+            description="Ticking or unticking this has no effect"),
+
+        scripts.String(
+            P_CSVSEP, optional=False, grouping="2.1",
+            description="Choose the csv separator.",
+            values=separators, default="TAB"),
+
+        scripts.Bool(
+            P_INCL_PARENT, optional=False,
+            grouping="2.2",
+            description="Check to include or not the name of the parent(s)" +
+                        " objects as columns in the csv", default=False),
+
+        scripts.Bool(
+            P_INCL_NS, optional=False,
+            grouping="2.3",
+            description="Check to include the annotation namespaces" +
+                        " in the csv file.", default=False),
+
+        scripts.Bool(
+            P_INCL_TAG, optional=False,
+            grouping="2.4",
+            description="Check to include tags in the csv file.",
+            default=False),
```

Comment:
For all these `Bool` parameters, the `optional=False` adds a red asterisk to the UI, but they are still optional (since I have to ignore them if I want to enter a False value). So the `optional=False` is kinda misleading/confusing.

Reply:
set optional to false for all bool parameters